Research, Workshop and Meetings, Minor Changes to multiple pages

krishnapandeykp opened the pull request:
Centres and Facilities - Removed Cover Image and reduced cover height Homepage - Changed News to Seminars and Announcements to News and Announcements, also binding the link to make it clickable. Truncated the title to consider the last word completely, rather than cutting it in between which fixed length substring does. And if no link is associated, remains on the same page. Changed the slides as well. **Media Coverage** - Changed the CSS mainly, to remove the unnecessary scrollbar, which was caused by (margin-left: 25vw), and in CSS we were restricting x-overflow. People - In mobile view, cards were not stacking, fixed that. And a few other minor changes. **Research** - Major Changes Done. Calling Project, Vignettes, Collaborations, and Overall Structure of Page. **StudentsSeminar** - Same Unwanted Scrollbar, x-overflow fixed. **Workshop and Meetings** - Workshop: How data appear, also removed the default <a> tag, because it was taking to the homepage if no link was present. Instead Using InnerHTML to call the link inside the Details directly. And Used Sanitizer for Security Purposes. Seminar: View of data appearance changed. reduced the height of the card by fixing a few unnecessary spaces. Also, how the Image appears in md and sm view. (found no other way better than this, like calling the image twice while blocking one at a time).

krishnapandeykp committed:
Research, Workshop and Meetings, Minor Changes to multiple pages

Centres and Facilities - Removed Cover Image and reduced cover height
Homepage - Changed News to Seminars and Announcements to News and Announcements, also binding the link to make it clickable. Truncated the title to consider the last word completely, rather than cutting it in between which fixed length substring does. And if no link is associated, remains on the same page. Changed the slides as well.
Media Coverage - Changed the CSS mainly, to remove the unnecessary scrollbar, which was caused by (margin-left: 25vw), and in CSS we were restricting x-overflow.
People - In mobile view, cards were not stacking, fixed that. And a few other minor changes.
Research - Major Changes Done. Calling Project, Vignettes, Collaborations, and Overall Structure of Page.
StudentsSeminar - Same Unwanted Scrollbar, x-overflow fixed.
Workshop and Meetings - Workshop: How data appear, also removed the default <a> tag, because it was taking to the homepage if no link was present. Instead Using InnerHTML to call the link inside the Details directly. And Used Sanitizer for Security Purposes.
Seminar: View of data appearance changed. reduced the height of the card by fixing a few unnecessary spaces. Also, how the Image appears in md and sm view. (found no other way better than this, like calling the image twice while blocking one at a time).

krishnapandeykp committed:
Merge branch 'development' of https://github.com/ICWaR/icwar.github.io into krishna

krishnapandeykp committed:
Merge branch 'development' into krishna

krishnapandeykp committed:
Merge branch 'krishna' of https://github.com/ICWaR/icwar.github.io into krishna

ICWaR commented on src/app/views/media-coverage/media-coverage.component.scss:
```diff
@@ -35,9 +35,9 @@ hr{
-  overflow-x: hidden;
-}
+// *{
+//   overflow-x: hidden;
```
Please explain this

krishnapandeykp replied:
Atually I took the code for this page from Workshop and Meeting Components, there also Overflow-x was there, so I didn't knew that it was creating problem, but once I found this was the only reason, I fixed it across. I commented out everywhere, just in case we need it later so. Once All will settle, will refine CSS once and will remove Commented Codes from everywhere.

ICWaR replied:
Okay, but from next time onwards please do not mark conversation as resolved by your own

ICWaR commented on src/app/views/research-new/research-new.component.html:
```diff
@@ -7,30 +7,140 @@
-    </ul>
+<div class="p-5 row w-100 bg-light">
+<div class="col-12 shadow px-5 p-2">
+  <h1 class="my-2 mb-5"><b>ReSearch : </b> Contents</h1>
```
S should not be in capital in word Research

krishnapandeykp replied:
Yeah Okay. I put it maybe to look creative but that's fine.

krishnapandeykp committed:
Research "S" has been corrected.

ICWaR commented on src/app/views/people/people.component.html:
```diff
@@ -9,39 +9,40 @@
     <div class="col-md-3 card shadow d-none d-md-table-row pb-5 ps-5 " style="width: 20vw;max-height: fit-content;position: sticky; ">
       <h2 class=" my-4 ">Contents</h2>
-      <ol class="facultySideNavHighlight card-body">
+      <ul class="facultySideNavHighlight card-body">
```
Explain why use ul here

krishnapandeykp replied:
![Screenshot 2024-03-26 165821](https://github.com/ICWaR/icwar.github.io/assets/138218874/e1535991-ab72-4919-aa6d-a66da511801b)


In your case, it was neither showing bullets nor Numbers, but once I did other changes, somehow it started showing Numbers, then I changed it to UL, SS attached for 'ol' case.

ICWaR replied:
We dont need numbers in here

ICWaR replied:
Also why did you change to ul can you explain that particularly.

krishnapandeykp replied:
![Screenshot 2024-03-26 184118](https://github.com/ICWaR/icwar.github.io/assets/138218874/c348d9c3-63ae-4268-88cc-e1d1dcb760c7)

This I achieved after using 'ul'. Prior given SS was the result of 'ol'.

ICWaR commented on src/app/views/people/people.component.html:
```diff
@@ -9,39 +9,40 @@
-  <div class="d-flex mt-5" style="margin-left: 25vw;width: 100vw;">
-    <h1 class="float-start">{{getTitleFromType(type)}} Directory</h1>
-  </div>
+  <div class="d-flex mt-5 " style="width: 100vw;">
```
Explain this

ICWaR replied:
Where is the explanation ??

krishnapandeykp replied:
Margin Left was Overwriting the usual 100vw, to take it beyond. Hence Overflow-x was required to curb that. Since I removed the Overflow-x case, I had to manage the overflow, so made the complete View Port size as 100vw.

ICWaR commented on src/app/views/people/people.component.scss:
```diff
@@ -31,9 +31,9 @@ div.d-flex{
-*{
-  overflow-x: hidden;
-}
+// *{
```
Did you check the output of your changes ??
<img width="831" alt="Screenshot 2024-03-26 at 4 08 02 PM" src="https://github.com/ICWaR/icwar.github.io/assets/154324075/d1eb1453-5ad4-480e-a329-e490bdfeab49">

krishnapandeykp replied:
On my system It still looks good, and in Mobile View as well. But in between the problem is there, This I'll fix.

![Screenshot 2024-03-26 170759](https://github.com/ICWaR/icwar.github.io/assets/138218874/40dabc5e-3a6b-4a84-831c-9057ead47ce6)

ICWaR replied:
Please fix this first. 
I had instructed earlier also that we have to test it for all types of screen sizes . Google how to check different screensizes in chrome and then change accordingly

krishnapandeykp replied:
![Screenshot 2024-03-26 201358](https://github.com/ICWaR/icwar.github.io/assets/138218874/2ba0121b-b6b5-470f-932d-b717a152c460)


Okay, Fixed it.

ICWaR commented on src/app/views/people/people.component.scss:
```diff
@@ -50,6 +50,7 @@ div.d-flex{
   // display: inline-block;
   position: relative;
   // color: #0087ca;
+  font-size: larger;
```
This is not to be set from here

krishnapandeykp replied:
Okay, I found it too small compared to the data present on the page. So, fixed it there. Should it be fixed from config.js?

ICWaR replied:
Yes, please do not change anything according to your choices. Will take proper acceptance first for every change.
Remove all the font-size changes from everyplace you have done and other similar changes

krishnapandeykp replied:
Okay, Only at two places I did, here in CSS and in Workshop Component- Inline. That also I changed now.

krishnapandeykp replied:
Only at two places I did it, once here and other one in Workshop Component, there it was Inline. I have Corrected it now.

ICWaR commented on src/app/views/centres-and-facilities/centres-and-facilities.component.html:
```diff
@@ -1,4 +1,4 @@
-<div style="height: 40vh; background-size: cover; background-image: url('https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTa3U1_-6QurYiV7Pd3a5F-DhYvfEM87eAtFKJHTGK4jh3ySppN2sTjb1qVWJNVKWlcVIw&usqp=CAU');">
+<div style="height: 20vh; background-size: cover;">
```
Check the output difference due to this

krishnapandeykp replied:
After removing the Background Image, it was looking too broad for just a line of text. So narrowed it down. Still, some problems need  to be fixed. Some, space on the right side is there.

![Screenshot 2024-03-26 171240](https://github.com/ICWaR/icwar.github.io/assets/138218874/acf71f29-d0a2-4aca-96a6-45a2676c26ac)

ICWaR replied:
So fix it first and then raise a pull request

krishnapandeykp replied:
Fixed it now.
![Screenshot 2024-03-26 185559](https://github.com/ICWaR/icwar.github.io/assets/138218874/bf6989fe-5277-4b8f-9edf-46b2906dc399)

ICWaR commented on src/app/views/homepage/homepage.component.html:
```diff
@@ -29,7 +29,7 @@ <h1 class="card-header">
       Chairman's Message</h1>
     <div class="card-body">
-      <p class="card-text d-none d-md-block" style="font-size: medium;;">
+      <p class="card-text d-none d-md-block" style="font-size: medium; text-align: justify;">
```
Explain why in all the files you have added text-align : justify
Is it needed?  If yes show image representation of the difference

krishnapandeykp replied:
![Screenshot 2024-03-26 171610](https://github.com/ICWaR/icwar.github.io/assets/138218874/09c99ae8-f961-4111-b9af-d98d0fbd5f40)
![Screenshot 2024-03-26 171621](https://github.com/ICWaR/icwar.github.io/assets/138218874/f66f0a43-fa40-419f-99ef-6ba0f14924a1)

Two SS, one without Justify, one after Justify. Without justify is there on icwar.github.io

ICWaR commented on src/app/views/workshops-and-meetings/workshops-and-meetings.component.ts:
```diff
@@ -1,60 +1,51 @@
-    });
+  // Sanitize the HTML content before binding it
+  getSanitizedDetails(details: string): SafeHtml {
+    return this.sanitizer.bypassSecurityTrustHtml(details);
```
Understand this first and then draft me an explanation why you have done this.

krishnapandeykp replied:
I do understand that in Our case, it may not be needed, as it's a static page, but then also, thought to use this to enhance the security case. It is mainly used to prevent Cross-Site Scripting (XSS) vulnerabilities. I thought I was using Inner HTML so you may question the security constraints. So, used this.

ICWaR replied:
You do not understand what you are doing. Please do not copy paste without understanding what it does.
First understand this line " this.sanitizer.bypassSecurityTrustHtml(details);". You have bypassed the checks .
Please think before you answer the next time

krishnapandeykp replied:
getSanitizedDetails(details: string): SafeHtml | null {
    if (details) {
      return this.sanitizer.sanitize(SecurityContext.HTML, details) || null;
    }
    return null;
  }
  
  fixed it by using the above code.
  
  Sorry for the oversight. Ofcourse It was new thing to me, but I'm not totally wrong here, I'm quoting the explanation I got, because of that only I put it there. 

"
    // Use DomSanitizer's bypassSecurityTrustHtml method to sanitize the HTML content
    return this.sanitizer.bypassSecurityTrustHtml(details);
    // This method returns a SafeHtml object, which represents the sanitized HTML content
    // It tells Angular that the HTML content is safe to render
"

krishnapandeykp committed:
Fixed the Font-size, Dom Sanitizer, and Email Overflow in People, C&F banner.

krishnapandeykp committed:
Research Collaborations Structure Updated and F&L redirected

krishnapandeykp committed:
Changes Done for Sidebar, and Fallback functionality of Seminars for Workshop Component has been improved.

Earlier if we opened the seminar and click on link to join, and if we switch back, it used to fall on workshops but now it remains on seminar page only.  But direct access to seminars section from navbar is yet to be achieved.

krishnapandeykp committed:
Data Update

architbubber committed:
Revert people page changes and Workshop changes

architbubber committed:
Working build committed
